app/models/spree/admin/resources/order_default_tabs_builder.rb

```diff
@@ -0,0 +1,217 @@
+            with_update_availability_check
+
+          root.add(tab)
+        end
+
+        def shipments_tab_config
+          {
+            icon_name: 'truck.svg',
+            name: :shipments,
+            url: ->(resource) { edit_admin_order_path(resource) },
+            classes: 'nav-link',
+            partial_name: :shipments
+          }
+        end
+
+        def add_adjustments_tab(root)
+          tab =
+            Tab.new(adjustments_tab_config).
+            with_active_check.
+            with_default_translator.
+            with_index_availability_check(::Spree::Adjustment)
+
+          root.add(tab)
+        end
+
+        def adjustments_tab_config
+          {
+            icon_name: 'adjust.svg',
+            name: :adjustments,
+            url: ->(resource) { admin_order_adjustments_path(resource) },
+            classes: 'nav-link',
+            partial_name: :adjustments
+          }
+        end
+
+        def add_payments_tab(root)
+          tab =
+            Tab.new(payments_tab_config).
+            with_active_check.
+            with_default_translator.
+            with_index_availability_check(::Spree::Payment)
+
+          root.add(tab)
+        end
+
+        def payments_tab_config
+          {
+            icon_name: 'credit-card.svg',
+            name: :payments,
+            url: ->(resource) { admin_order_payments_path(resource) },
+            classes: 'nav-link',
+            partial_name: :payments
+          }
+        end
+
+        def add_return_authorizations_tab(root)
+          tab =
+            Tab.new(return_authorizations_tab_config).
+            with_active_check.
+            with_default_translator.
+            with_index_availability_check(::Spree::ReturnAuthorization)
+
+          root.add(tab)
+        end
+
+        def return_authorizations_tab_config
+          {
+            icon_name: 'enter.svg',
+            name: :return_authorizations,
+            url: ->(resource) { admin_order_return_authorizations_path(resource) },
+            classes: 'nav-link',
+            partial_name: :return_authorizations
+          }
+        end
+
+        def add_customer_returns_tab(root)
+          tab =
+            Tab.new(customer_returns_tab_config).
+            with_active_check.
+            with_default_translator.
+            with_index_availability_check(::Spree::CustomerReturn)
+
+          root.add(tab)
+        end
+
+        def customer_returns_tab_config
+          {
+            icon_name: 'returns.svg',
+            name: :customer_returns,
+            url: ->(resource) { admin_order_customer_returns_path(resource) },
+            classes: 'nav-link',
+            partial_name: :customer_returns
+          }
+        end
+
+        def add_state_changes_tab(root)
+          tab =
+            Tab.new(state_changes_tab_config).
+            with_active_check.
+            with_custom_translator(::Spree::StateChange, :human_attribute_name).
```

Comment:
Hm, looking at this one, it's a bit weird. The tab is called "State changes" anyway, it seems that we're just missing a translation in the i18n file and someone decided to use some active support magic to generate a label from class name.  Maybe instead of copying this, we could just add the correct key to the translations file and use it?

Reply:
Good catch! 😃
I've submitted a PR to `spree/core` with missing translations - https://github.com/spree/spree/pull/11938
Having this merged renders the `Translator` module unnecessary (there's no better code than no code, right? 😉)